tests/test_shoplist_unit.py

```
@@ -0,0 +1,237 @@
+    assert generator._convert_to_grams(100, "g") == 100
+    assert generator._convert_to_grams(1, "kg") == 1000
+    assert generator._convert_to_grams(250, "ml") == 250
+    assert generator._convert_to_grams(2, "l") == 2000
+    # Heuristics
+    assert generator._convert_to_grams(3, "pcs") == 300
+    assert generator._convert_to_grams(2, "tbsp") == 30
+    assert generator._convert_to_grams(3, "tsp") == 15
+    assert generator._convert_to_grams(1, "cup") == 250
+
+
+def test_packaging_rules_load_error_fallback(tmp_path: Path) -> None:
```

Flake8 (7.3.0)

[error] undefined name 'Path'

(F821)